src/storage-client/src/sink.rs

```diff
@@ -283,18 +286,22 @@ pub async fn build_kafka(
             )
             .await?;
 
+        let progress_client = Arc::new(progress_client);
         let latest_ts = determine_latest_progress_record(
             format!("build_kafka_{}", sink_id),
             progress_topic.name().to_string(),
             ProgressKey::new(sink_id),
-            Arc::new(progress_client),
+            Arc::clone(&progress_client),
         )
-        .await?;
+        .await
+        .check_ssh_status(progress_client.client().context())?;
```

Comment:
Is it always valid to override the error with the SSH error? I guess so? Like, if the SSH tunnel is broken, you're screwed, even if there was another error..

Reply:
Yeah, thats the idea!